lib/iris/fileformats/netcdf/loader.py

```diff
@@ -232,18 +239,33 @@ def _get_cf_var_data(cf_var, filename):
             chunks = cf_var.cf_data.chunking()
             # In the "contiguous" case, pass chunks=None to 'as_lazy_data'.
             if chunks == "contiguous":
-                chunks = None
-
-            # Return a dask array providing deferred access.
-            result = as_lazy_data(proxy, chunks=chunks)
-
-    return result
+                # Equivalent to chunks=None, but value required by chunking control
+                chunks = list(cf_var.shape)
+
+            # Modify the chunking in the context of an active chunking control.
+            # N.B. settings specific to this named var override global ('*') ones.
+            dim_chunks = CHUNK_CONTROL.var_dim_chunksizes.get(
+                cf_var.name
+            ) or CHUNK_CONTROL.var_dim_chunksizes.get("*")
+            if not dim_chunks:
+                dims_fixed = None
+            else:
+                # Modify the chunks argument, and pass in a list of 'fixed' dims, for
+                # any of our dims which are controlled.
+                dims = cf_var.cf_data.dimensions
+                dims_fixed = np.zeros(len(dims), dtype=bool)
+                for i_dim, dim_name in enumerate(dims):
+                    dim_chunksize = dim_chunks.get(dim_name)
+                    if dim_chunksize:
+                        chunks[i_dim] = dim_chunksize
+                        dims_fixed[i_dim] = True
+
+            return as_lazy_data(proxy, chunks=chunks, dims_fixed=dims_fixed)
```

Comment:
Since #4572 was written, `_get_cf_var_data()` no longer has a single `return` statement. It now uses `return result`, having assigned a value to `result` (different values assigned in different `if` blocks).

We didn't adjust the additions to `_get_cf_var_data()` to account for this new approach.